src/Queue/TenantAwareJob.php

```diff
@@ -56,4 +58,16 @@ public function __wakeup()
 
         $this->serializedWakeup();
     }
+
+    /**
+     * Manually override the hostname to be used.
+     *
+     * @param mixed $hostname
```

Comment:
I'm not a fan of mixed. Proper type hinting is crucial to get a good experience in your IDE. I agree we should allow both the Hostname object and an Id to be used as argument, if that's the case make sure the phpdoc matches.

Also this phpdoc is missing a `@return`.